syntax = "proto2";

package ru.sktbelpa.Productomer_2;


enum INFO {
    // верися этого протокола
    PROTOCOL_VERSION = 1;

    // идентификатор устройства РЧ-2_2
    PRODUCTOMER_2_ID = 0x0000E002;

    // ИД, преднозначенный для поиска
    ID_DISCOVER = 0xFFFF;
}

//------------------------------------------------------------------------------

// код выполнения операции
enum STATUS {
    // успешное завершение
    OK = 0;
    // недопустимое значение поля
    ERRORS_IN_SUBCOMMANDS = 1;

    PROTOCOL_ERROR = 100;
}

//------------------------------------------------------------------------------

// Общий запрос: Вызывающий всегда посылает только этот запрос, который содержит
// все возможные данные как необязательные поля.
message Request {
    // Идентификатор запроса. Служит для определения пары запрос/ответ
    required uint32 id = 1;
    // id устройства (идентификация)
    required uint32 deviceID = 2;
    // версия протокола запрашивающего
    required uint32 protocolVersion = 3;

    // Таймштамп клиента. Если он есть в сообщении, то часы устройства будут установлены в соответствии с ним
    optional fixed64 setClock = 4;
}

//------------------------------------------------------------------------------

// Общий ответ: Устройство всегда отвечает только этим ответом, который содержит
// все возможные данные как необязательные поля.
message Response {
    // Идентификатор запроса, которому соответствует этот ответ
    required uint32 id = 1;
    // id устройства (идентификация)
    required uint32 deviceID = 2;
    // версия протокола
    required uint32 protocolVersion = 3;
    // общий статус выполнения
    required STATUS Global_status = 4;
    // таймштамп устройства, когда фактически был сгененрирован ответ
    required fixed64 timestamp = 5;
}

//------------------------------------------------------------------------------
